src/main/java/com/puppycrawl/tools/checkstyle/checks/regexp/RegexpCheck.java

```diff
@@ -304,69 +295,59 @@ public int[] getRequiredTokens() {
     @SuppressWarnings("deprecation")
     @Override
     public void beginTree(DetailAST rootAST) {
-        matcher = format.matcher(getFileContents().getText().getFullText());
-        matchCount = 0;
-        errorCount = 0;
-        findMatch();
+        final Matcher matcher = format.matcher(getFileContents().getText().getFullText());
+        processRegexpMatches(matcher);
     }
 
     /**
-     * Recursive method that finds the matches.
+     * Finds the matches in the file.
```

Comment:
This documentation is misleading. Could you update it to reflect what the method actually does?

Reply:
Processes the regexp matches and finds the number of errors in the file. is this better ?

Reply:
yes, it's better

Reply:
done